test/plugins/http.spec.js

```diff
@@ -11,444 +11,520 @@ describe('Plugin', () => {
+              const req = http.request(`${protocol}://localhost:${port}/user`, res => {
                 expect(tracer.scopeManager().active()).to.be.null
                 done()
               })
-            })
 
-            req.end()
+              req.end()
+            })
           })
         })
-      })
 
-      it('should handle errors', done => {
-        getPort().then(port => {
-          agent
-            .use(traces => {
-              expect(traces[0][0].meta).to.have.property('error.type', 'Error')
-              expect(traces[0][0].meta).to.have.property('error.msg', `connect ECONNREFUSED 127.0.0.1:${port}`)
-              expect(traces[0][0].meta).to.have.property('error.stack')
+        it('should run the event listeners in the parent context', done => {
+          const app = express()
+
+          app.get('/user', (req, res) => {
+            res.status(200).send('OK')
+          })
+
+          getPort().then(port => {
+            appListener = server(app, port, () => {
+              const req = http.request(`${protocol}://localhost:${port}/user`, res => {
+                res.on('data', () => {})
+                res.on('end', () => {
+                  expect(tracer.scopeManager().active()).to.be.null
+                  done()
+                })
+              })
+
+              req.end()
             })
-            .then(done)
-            .catch(done)
+          })
+        })
 
-          const req = http.request(`http://localhost:${port}/user`)
+        it('should handle errors', done => {
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                expect(traces[0][0].meta).to.have.property('error.type', 'Error')
+                expect(traces[0][0].meta).to.have.property('error.msg', `connect ECONNREFUSED 127.0.0.1:${port}`)
+                expect(traces[0][0].meta).to.have.property('error.stack')
+              })
+              .then(done)
+              .catch(done)
 
-          req.end()
+            const req = http.request(`${protocol}://localhost:${port}/user`)
+
+            req.end()
+          })
         })
-      })
-    })
 
-    describe('with service configuration', () => {
-      let config
+        it('should only record a request once', done => {
+          const app = express()
 
-      beforeEach(() => {
-        config = {
-          service: 'custom'
-        }
+          app.get('/user', (req, res) => {
+            res.status(200).send()
+          })
+
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                const spans = traces[0]
+                expect(spans.length).to.equal(3)
+              })
+              .then(done)
+              .catch(done)
+
+            appListener = server(app, port, () => {
+              // Activate a new parent span so we capture any double counting that may happen
+              const span = tracer.startSpan('http-test')
```

Comment:
Why is this needed?

Reply:
Because the two spans that are being emitted in the current release are siblings, thus the test would stop after the first one came in. Adding a parent span for them both makes sure the test takes all emitted spans into consideration.